src/Esfa.Vacancy.Infrastructure/Caching/AzureRedisCacheService.cs

```diff
@@ -9,17 +9,18 @@
 
 namespace Esfa.Vacancy.Infrastructure.Caching
 {
-
     public class AzureRedisCacheService : ICacheService
     {
+        private const int DefaultCacheDurationHours = 4;
         private readonly Func<ConnectionMultiplexer> _connectionFactory;
         private readonly ILog _logger;
         private ConnectionMultiplexer _connection;
-
+        private readonly string _cacheDuration;
 
         public AzureRedisCacheService(IProvideSettings settings, ILog logger)
         {
             _logger = logger;
+            _cacheDuration = settings.GetSetting(ApplicationSettingKeys.CacheReferenceDataDuration);
```

Comment:
Can we not have any code in constructor that could fail resolution of the object?